frame/offences/src/lib.rs

```diff
@@ -80,11 +90,22 @@ decl_event!(
 );
 
 decl_module! {
-	/// Offences module, currently just responsible for taking offence reports.
 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
 		fn deposit_event() = default;
+
+		fn on_initialize(_now: T::BlockNumber) {
+			// only decode storage if we can actually submit anything again.
+			if T::OnOffenceHandler::can_report() {
+				// defensive only. Per definition, `OnOffenceHandler` should accept reports when
+				// `can_report` is true.
+				<DeferredOffences<T>>::take()
+					.iter()
+					.for_each(|(o, p, s)| Self::report_or_store_offence(o, p, *s));
```

Comment:
Should we print a warning at least if it wasn't accepted twice?

Reply:
Yes that'd be good 👍